Makefile

```diff
@@ -0,0 +1,74 @@
+# Go parameters
+GOCMD	= go
+GOBUILD	= $(GOCMD) build
+GOCLEAN	= $(GOCMD) clean
+GOTEST	= $(GOCMD) test
+GOTOOL	= $(GOCMD) tool
+GOGET	= $(GOCMD) get
+GOMOD	= $(GOCMD) mod
+GOINST	= $(GOCMD) install
+GORUN	= ${GOCMD} run
+
+BUILD_DIR		:= build
+TEST_BUILD_DIR  := build/test
+BINARY_NAME		:= nginx-agent
+BINARY			:= $(BUILD_DIR)/$(BINARY_NAME)
+PROJECT_DIR		= cmd/agent
+PROJECT_FILE	= main.go
+IMPORT_MAPPING 	:= ../common/common.yaml:github.com/nginx/agent/v3/internal/api/http/common
+
```

Comment:
Had a Zoom discussion about this with Donal. I'll add the same explanation that I gave there here too for the purpose of a `.PHONY`.

So a `.PHONY` is meant to let `make` know that the target does not produce a directory or file. Essentially every target that is defined in a Makefile is by default supposed to produce a directory or file that is the same name as the target. 

e.g. the target `clean` is by default expected to produce a directory called `clean`. Now when you add `clean` to `.PHONY`, you are saying that `clean` is essentially a function that does something instead of producing the `clean` directory.

In case my rambling explanation wasn't clear, here's maybe a better one: https://stackoverflow.com/questions/2145590/what-is-the-purpose-of-phony-in-a-makefile